scripts/data_files/error.fmt

```diff
@@ -42,10 +42,66 @@
 
 HEADER_INCLUDED
 
+typedef struct mbedtls_error
+{
+    int code;                   /* Error code. */
+    const char * description;   /* Error description. */
+} mbedtls_error_t;
+
+static mbedtls_error_t high_level_errors[] =
+{
+HIGH_LEVEL_CODE_CHECKS
+};
+
+#define NUM_HIGH_LEVEL_ERRORS ( sizeof(high_level_errors)/sizeof(mbedtls_error_t) )
+
+static mbedtls_error_t low_level_errors[] =
+{
+LOW_LEVEL_CODE_CHECKS
+};
+
+#define NUM_LOW_LEVEL_ERRORS  ( sizeof(low_level_errors)/sizeof(mbedtls_error_t) )
+
+const char * mbedtls_high_level_strerr( int error_code )
+{
+    size_t i;
+    const char *error_description = NULL;
+
+    for(i = 0; i < NUM_HIGH_LEVEL_ERRORS; i++ )
+    {
+        if( high_level_errors[i].code == error_code )
+        {
+            error_description = high_level_errors[i].description;
+            break;
+        }
+    }
+
+    return error_description;
+}
+
+const char * mbedtls_low_level_strerr( int error_code )
+{
+    size_t i;
+    const char *error_description = NULL;
+
+    for(i = 0; i < NUM_LOW_LEVEL_ERRORS; i++ )
+    {
+        if( low_level_errors[i].code == error_code )
```

Comment:
Why traverse a constant data structure? Why not  `switch(error_code)`? Wouldn't that result in smaller code by giving the compiler more room to optimize?

Reply:
Agreed that it will be better to use ```switch case``` as it will result in a faster code. Made the change.